### PR TITLE
chore: re-scaffold project

### PR DESCRIPTION
**Pull request by arturshadnik**
## Issue
#297 

## Description
This PR re-scaffolds the project using `kubebuilder alpha generate`. The 2 major changes are:
- Remove all usage of `kube-rbac-proxy` and replace with [WithAuthenticationAndAuthorization ](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/metrics/filters#WithAuthenticationAndAuthorization) due to `kube-rbac-proxy` being deprecated
- Initialize e2e tests. For now, we are only testing basic functionality. The tests should be expanded to cover validator-specific use-cases.

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -4,5 +4,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/validator-labs/validator-plugin-network
-  newTag: latest
+  newName: example.com/validator-plugin-network
```
Comment (mattwelke) [inline config/manager/kustomization.yaml:7]: Just checking... do we want "example" here?

**Review by mattwelke — COMMENTED** [review]
Overall, LGTM. Just pointed out one thing though.

File: cmd/main.go [hunk 2]
```diff
@@ -46,18 +47,28 @@ var (
 
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
-	utilruntime.Must(validatorv1alpha1.AddToScheme(scheme))
```
Comment (TylerGillson) [inline cmd/main.go:49]: Keep this please

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (arturshadnik) [reply config/manager/kustomization.yaml:7]: Thanks, good catch. no...

**Pull request merged**